engine/server/engine/server/engine_connect_server_service.go

```diff
@@ -0,0 +1,244 @@
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred while cleaning enclaves")
+	}
+
+	response := &kurtosis_engine_rpc_api_bindings.CleanResponse{RemovedEnclaveNameAndUuids: removedEnclaveUuidsAndNames}
+	return connect_go.NewResponse(response), nil
+}
+
+func (service *EngineConnectServerService) GetServiceLogs(ctx context.Context, connectArgs *connect_go.Request[kurtosis_engine_rpc_api_bindings.GetServiceLogsArgs], stream *connect_go.ServerStream[kurtosis_engine_rpc_api_bindings.GetServiceLogsResponse]) error {
+	args := connectArgs.Msg
+	enclaveIdentifier := args.GetEnclaveIdentifier()
+	enclaveUuid, err := service.enclaveManager.GetEnclaveUuidForEnclaveIdentifier(context.Background(), enclaveIdentifier)
+	if err != nil {
+		logrus.Errorf("An error occurred while fetching uuid for enclave '%v'. This could happen if the enclave has been deleted. Treating it as UUID", enclaveIdentifier)
+		enclaveUuid = enclave.EnclaveUUID(enclaveIdentifier)
+	}
+	serviceUuidStrSet := args.GetServiceUuidSet()
+	requestedServiceUuids := make(map[user_service.ServiceUUID]bool, len(serviceUuidStrSet))
+	shouldFollowLogs := args.FollowLogs
+
+	for serviceUuidStr := range serviceUuidStrSet {
+		serviceUuid := user_service.ServiceUUID(serviceUuidStr)
+		requestedServiceUuids[serviceUuid] = true
+	}
+
+	if service.logsDatabaseClient == nil {
+		return stacktrace.NewError("It's not possible to return service logs because there is no logs database client; this is bug in Kurtosis")
+	}
+
+	var (
+		serviceLogsByServiceUuidChan chan map[user_service.ServiceUUID][]logline.LogLine
+		errChan                      chan error
+		cancelCtxFunc                func()
+	)
+
+	notFoundServiceUuids, err := service.reportAnyMissingUuidsAndGetNotFoundUuidsList(ctx, enclaveUuid, requestedServiceUuids, stream)
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred reporting missing user service UUIDs for enclave '%v' and requested service UUIDs '%+v'", enclaveUuid, requestedServiceUuids)
+	}
+
+	conjunctiveLogLineFilters, err := newConjunctiveLogLineFiltersFromGRPCLogLineFilters(args.GetConjunctiveFilters())
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred creating the conjunctive log line filters from the GRPC's conjunctive log line filters '%+v'", args.GetConjunctiveFilters())
+	}
+
+	serviceLogsByServiceUuidChan, errChan, cancelCtxFunc, err = service.logsDatabaseClient.StreamUserServiceLogs(ctx, enclaveUuid, requestedServiceUuids, conjunctiveLogLineFilters, shouldFollowLogs)
+	if err != nil {
+		return stacktrace.Propagate(
+			err,
+			"An error occurred streaming service logs for UUIDs '%+v' in enclave with ID '%v' using filters '%v+' "+
+				"and with should follow logs value as '%v'",
+			requestedServiceUuids,
+			enclaveUuid,
+			conjunctiveLogLineFilters,
+			shouldFollowLogs,
+		)
+	}
+	defer func() {
+		if cancelCtxFunc != nil {
+			cancelCtxFunc()
+		}
+	}()
+
+	for {
```

Comment:
I assume all this logic in this file is mostly copied from engine_server_service.go, is that correct? Haven't reviewed the code thoroughly here

Reply:
yes; i have updated the how we handle request and response; the only thing to look for is streams.. There is no notion of stream.Context() so needed to change that a bit for `GetServiceLogs`